ClangSharpPInvokeGenerator/Cursors/Exprs/UnaryOperator.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Diagnostics;
+using ClangSharp;
+
+namespace ClangSharpPInvokeGenerator
+{
+    internal sealed class UnaryOperator : Expr
+    {
+        private readonly Lazy<bool> _isPrefix;
+        private readonly Lazy<string> _operator;
+
+        private Expr _expr;
+
+        public UnaryOperator(CXCursor handle, Cursor parent) : base(handle, parent)
+        {
+            Debug.Assert(handle.Kind == CXCursorKind.CXCursor_UnaryOperator);
+
+            _isPrefix = new Lazy<bool>(() => {
+                switch (Operator)
+                {
+                    case "-":
+                    {
+                        return true;
+                    }
+
+                    default:
+                    {
+                        Debug.WriteLine($"Unhandled operator kind: {Operator}.");
+                        Debugger.Break();
+                        return false;
+                    }
+                }
+            });
+
+            _operator = new Lazy<string>(() => {
+                var tokens = TranslationUnit.Tokenize(this);
+
+                Debug.Assert(tokens.Length >= 2);
+
+                var operatorIndex = GetOperatorIndex(tokens);
+                Debug.Assert(tokens[operatorIndex].Kind == CXTokenKind.CXToken_Punctuation);
+                return tokens[operatorIndex].GetSpelling(Handle.TranslationUnit).ToString();
+            });
+        }
+
+        public Expr Expr
+        {
+            get
+            {
+                return _expr;
+            }
+
+            set
+            {
+                Debug.Assert(_expr is null);
```

Comment:
We are also able to do additional sanity checks, such as an `UnaryOperator` only having a single expression (which can itself have sub-expressions).